style-guide.rst

```diff
@@ -0,0 +1,628 @@
+  
+The word *tag* has often been used to refer to the entire element. For clarity, we will avoid that here.
+
+
+.. _xml-node:
+
+Node
+~~~~~
+
+The word *node* is often used interchangeably with *element*.
+
+For clarity, we make the following distinction:
+
+- An HTML or XML document has *elements*, not *nodes*.
+- A *node* is part of a "live" DOM tree or other dynamic representation.
+
+  - An XML or HTML element becomes an *element node* in a DOM tree.
+  - There are also other types of nodes in a DOM tree.
+
+.. _xml-attributes-values:
+
+Attributes and values
+~~~~~~~~~~~~~~~~~~~~~~~
+
+An element may have attributes. Attributes have values. Values are wrapped in straight double-quotes.
+
+.. code-block:: xml
+
+  <element attribute="value">
+    Content.
+  </element>
+  
+Other names for attributes, such as *variables* or *properties*, should be avoided.
+
+.. _xml-element-content:
+  
+Element content
+~~~~~~~~~~~~~~~~
+
+The code between the opening and closing tags of an element is the content. Content can include other elements, which are called *child elements*.
+
+.. code-block:: xml
+
+  <element>
+    Content.
+    <child-element>
+      More content.
+    </child-element>
+  </element>
+  
+When an element is empty, it can be called a *null element*.
+
+.. code-block:: xml
+
+  <null-element attribute="value" />
+
+In XML, null element tags always self-close. This is not the case in HTML. 
+
+- HTML elements that are always null (for example, `<img>`) do not need to be self-closed.
+- Empty HTML elements that normally accept content have a separate closing tag.
+
+.. code-block:: html
+
+  <img src="awesome-picture.jpeg">
+
+  <script src="some-javascript.js"></script>
+
+.. _xml-capitalization:
+  
+Capitalization
+~~~~~~~~~~~~~~~~
+
+For all HTML samples, tag names and attribute names should be ``all lowercase``. 
+
+Newly-written XML examples should also be ``all lowercase``.
+
+XML examples that show actual code generated by tools in the ODK ecosystem should replicate that code exactly, regardless of its capitalization practice.
+
+.. _odk-jargon:
+
+ODK jargon
+=============
+
+.. _writing-about-odk:
+
+ODK and ODK Docs
+-------------------
+
+.. rubric:: Wrong
+
+- Odk
+- odk
+- Open data kit
+- OpenDataKit
+- ODK docs
+- ODK documentation
+
+.. rubric:: Right
+
+- ODK
+- Open Data Kit
+- ODK Docs
+- ODK Documentation
+
+.. rubric:: Probably want to avoid...
+
+- Open Data Kit Documentation
+
+.. _odk-app-project-names:
+
+ODK app and project names
+---------------------------
+
+ODK includes a number of components, including:
+
+- Collect
+- Aggregate
+- Briefcase
+
+These should always be capitalized.
+
+The **ODK** prefix (as in, *ODK Collect*) should be used the first time a document mentions the app or project, or any other time it would be unclear.
+
+A few projects should *always* use the **ODK** prefix:
+
+- ODK XForm
+- ODK Javarosa
+- ODK Docs
+
+.. _xform-xlsform:
+
+XForm and XLSForm
```

Comment:
It's XForms (with an s). An individual form is called an XForm. 

XLSForm, confusingly, doesn't have an s. It probably should, but that ship has sailed...